File: botnanajs/examples/programming/p1.js
"use strict";

var botnana = require("../../index");

function test() {
  // Show reponse data
  botnana.debug_level = 1;
  botnana.once("ready", function() {
    // Programming API
    var Program = botnana.Program;
    // Program p1
    var p1 = new Program("p1");
    p1.deploy();
    // Run Program p1
    botnana.once("deployed", function() {
      p1.run();
    });
  });
  botnana.start("ws://192.168.7.2:3012");
}

test();

File: botnanars/examples/p1.rs
extern crate botnanars;
use botnanars::Botnana;
use std::sync::{Arc, Mutex};

fn main() {
    let botnana = Arc::new(Mutex::new(Botnana::new().unwrap()));

    let btn = botnana.clone();
    botnana.lock().unwrap().once("ready", move |_| {
        let mut p = btn.lock().unwrap().program("p1");
        p.deploy();
    });

    let btn = botnana.clone();
    botnana.lock().unwrap().once("deployed", move |_| {
        let p = btn.lock().unwrap().program("p1");
        p.run();
    });

    botnana.lock().unwrap().start("ws://192.168.50.197:3012");
    loop {}
}
